### PR TITLE
ST-1214: implement in-stock availability informer

### PR DESCRIPTION
**Pull request by goldenmaya**
Description:
https://virtocommerce.atlassian.net/browse/ST-1214

--

QA-test: ST-1217

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.3.0-pr-187-36e10b0c.zip

### CODE REVIEW

File: client-app/components/molecules/in-stock/in-stock.vue [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<template>
```
Comment (Flanker72) [inline client-app/components/molecules/in-stock/in-stock.vue:1]: This should be atom, not molecule. The markup consists from only HTML tags, there are no other atoms here.

File: client-app/pages/product/product.vue [hunk 2]
```diff
@@ -48,6 +48,12 @@
 
             <div class="mt-7 md:mt-5">
               <AddToCart :product="product" />
+
+              <VcInStock
+                :isInStock="product.availabilityData?.isInStock"
```
Comment (Flanker72) [inline client-app/pages/product/product.vue:53]: We use kebab-case for props, so please, use `:is-in-stock` here.

File: client-app/ui-kit/components/atoms/in-stock/in-stock.vue [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div
+    v-if="isInStock"
+    class="py-[1px] px-[0.625rem] bg-[color:var(--color-in-stock-available-bg)] text-[color:var(--color-in-stock-available)] whitespace-nowrap rounded-full text-[13px] leading-5"
```
Comment (Flanker72) [inline client-app/ui-kit/components/atoms/in-stock/in-stock.vue:4]: In the design there is `11px` text size

File: client-app/shared/catalog/components/product-card-list.vue [hunk 4]
```diff
@@ -43,15 +43,11 @@
...
-          {{ $t("common.suffixes.product_count_in_stock") }}
-        </div>
-        <div v-else class="flex items-center text-[color:var(--color-danger)] text-xs mt-3">
-          <div class="w-1.5 h-1.5 bg-[color:var(--color-danger)] rounded mr-1"></div>
-          {{ $t("common.messages.product_out_of_stock") }}
-        </div>
+        <VcInStock
+          :is-in-stock="product.availabilityData?.isInStock"
+          :quantity="product.availabilityData?.availableQuantity"
+          class="inline-block mt-2.5"
```
Comment (Flanker72) [inline client-app/shared/catalog/components/product-card-list.vue:49]: Too much space for list view

**Review by Flanker72 — APPROVED** [review]

File: client-app/ui-kit/components/atoms/in-stock/in-stock.vue [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<template>
```
Comment (trueboroda) [inline client-app/ui-kit/components/atoms/in-stock/in-stock.vue:1]: Add vc prefix to file. For consistency.

**Review by trueboroda — APPROVED** [review]

**Pull request merged**